vscode-dotnet-runtime-library/src/Acquisition/GlobalInstallerResolver.ts

```diff
@@ -0,0 +1,351 @@
+            const indexUrl = this.getIndexUrl(VersionResolver.getMajorMinor(this.fullySpecifiedVersionRequested));
+            return this.findCorrectInstallerUrl(this.fullySpecifiedVersionRequested, indexUrl);
+        }
+
+        const err = new DotnetVersionResolutionError(new Error(`The requested version resolved version is invalid.`), version);
+        this.eventStream.post(err);
+        throw err;
+    }
+
+    /**
+     *
+     * @remarks this function handles finding the right os, arch url for the installer.
+     * @param specificVersion the full, specific version, e.g. 7.0.301 to get.
+     * @param indexUrl The url of the index server that hosts installer downlod links.
+     * @returns The installer url to download.
+     */
+    private async findCorrectInstallerUrl(specificVersion : string, indexUrl : string) : Promise<string>
+    {
+        if(specificVersion === null || specificVersion === undefined || specificVersion === '')
+        {
+            const versionErr = new DotnetVersionResolutionError(new Error(`The requested version resolved version is invalid.`), specificVersion);
+            this.eventStream.post(versionErr);
+            throw versionErr;
+        }
+
+        const operatingSys : string = os.platform();
+        const operatingArch : string = os.arch();
+
+        let convertedOs = '';
+        let convertedArch = '';
+
+        switch(operatingSys)
+        {
+            case 'win32': {
+                convertedOs = 'win';
+                break;
+            }
+            case 'darwin': {
+                convertedOs = 'osx';
+                break;
+            }
+            case 'linux': {
+                convertedOs = operatingSys;
+                break;
+            }
+            default:
+            {
+                const osErr = new DotnetUnexpectedInstallerOSError(new Error(`The OS ${operatingSys} is currently unsupported or unknown.`));
+                this.eventStream.post(osErr);
+                throw osErr;
+            }
+        }
+
+        switch(operatingArch)
+        {
+            case 'x64': {
+                convertedArch = operatingArch;
+                break;
+            }
+            case 'x32': {
+                convertedArch = 'x86';
+                break;
+            }
+            case 'arm': {
+                convertedArch = operatingArch;
+                break;
+            }
+            case 'arm64': {
+                convertedArch = operatingArch;
+                break;
+            }
+            default:
+            {
+                const archErr = new DotnetUnexpectedInstallerOSError(new Error(`The architecture ${operatingArch} is currently unsupported or unknown.`));
+                this.eventStream.post(archErr);
+                throw archErr;
+            }
+        }
+
+        const desiredRidPackage = `${convertedOs}-${convertedArch}`;
+
+        const indexJson =  await this.fetchJsonObjectFromUrl(indexUrl);
```

Comment:
Hm, that's a really good point and I had to think about it for a bit. All of that is in `VersionResolver`, and uses the releases-index, which is a different format and url than the releases.json. This logic is fairly coupled to the specifics of the GlobalInstallerResolver, so I don't really think it would be the cleanest to extract it out. 

With that said, I think I should go ahead and move that json object function into the web worker code :)